drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -172,6 +173,67 @@ def check_mutually_exclusive_attributes(document)
     link_check.call(document)
   end
 
+  def check_condition(condition, item_hash, link_id)
+    return unless condition.key?('question') # Not currently validating conditions with local constants; see below
+
+    # Find the referenced question.
+    # item_hash does not contain groups, and the referenced question could be a group (with an EXISTS condition, for instance).
+    # But for now we are only validating against pick_list_options and pick_list_reference (which groups won't have).
+    # Therefore, if item_hash doesn't contain the referenced question, just return without raising an error.
+    referenced_question = item_hash[condition['question']]
+    return if referenced_question.nil?
+    return unless referenced_question['pick_list_options'] || referenced_question['pick_list_reference']
+
+    if referenced_question['pick_list_reference']
+      valid_answer_codes = pick_list_reference_to_allowed_values[referenced_question['pick_list_reference']]
+
+      # See comments on pick_list_reference_to_allowed_values.
+      # If options is nil, this picklist falls under case #1, and we don't attempt to validate its dependent questions.
+      return if valid_answer_codes.nil?
+    else
+      valid_answer_codes = referenced_question['pick_list_options'].map { |opt| opt['code'].to_s }
+      valid_answer_group_codes = referenced_question['pick_list_options'].map { |opt| opt['group_code'].to_s }.uniq
+    end
+
+    answer_codes = condition.values_at('answer_code', 'answer_codes').flatten.compact.uniq
+    if answer_codes.any?
+      (answer_codes - valid_answer_codes).each do |code|
+        add_issue("Invalid answer code: #{code} in 'enable_when' prop of #{link_id}")
+      end
+    end
+
+    if condition.key?('answer_group_code') && valid_answer_group_codes # rubocop:disable Style/GuardClause
+      code = condition['answer_group_code']
+      add_issue("Invalid answer group code: #{code} in 'enable_when' prop of #{link_id}") unless valid_answer_group_codes.include?(code)
+    end
+
+    # TODO: Additional validations. We attempt to ensure this validity in the form property editor,
+    # but we do not validate it here. For example:
+    # - if source is a question, the answer field should be compatible with the question type (eg shouldn't compare STRING=DATE)
+    # - if source is a local constant, the answer field should be compatible local constant type (eg shouldn't compare STRING=DATE)
+    # - if operator is special boolean operator (EXISTS/ENABLED), then the answer type should always be boolean
+    # - certain comparison operators should only be used for certain question types (eg can't use LESS_THAN on a STRING type)
+  end
+
+  def check_conditions(document, item_hash)
```

Comment:
```suggestion
  # Check that all EnableWhen and AutofillWhen conditions that reference specific Choice option codes are valid.
  def check_conditions(document, item_hash)
```

Helpful to have a comment to differentiate from `check_references` method which  just checks that the link_id references are valid. Maybe adding a header comment to that method as well